config/config.go

```diff
@@ -25,7 +26,7 @@ func FromYAMLFile(name string, v Validator) error {
 		return errors.Wrapf(ErrInvalidArgument, "non-nil pointer expected, got %T", v)
 	}
 
-	f, err := os.Open(name)
+	f, err := os.Open(filepath.Clean(name))
```

Comment:
Okay, but how or why does this please the linter?

Based on [filepath.Cleans documentation](https://pkg.go.dev/path/filepath#Clean) I cannot see what security this might bring? This should only sanitize the path, but shouldn't change the target.

Reply:
The `gosec` reports this as an error and causes the GitHub actions to fail, so I had to either fix it or suppress it.
```bash
[/Users/yhabteab/Workspace/go/icinga-go-library/config/config.go:28] - G304 (CWE-22): Potential file inclusion via variable (Confidence: HIGH, Severity: MEDIUM)
    27: 
  > 28:         f, err := os.Open(name)
    29:         if err != nil {
```

Reply:
It's complaining about this issue: https://securego.io/docs/rules/g304.html, but it doesn't seem to get a bit more context in which such an issue could be dangerous.

Reply:
Yes, but just adding a `filepath.Clean` call does not solve the addressed issue. This would also require the prefix check as demonstrated in the "The right way" code listing on your URL. Thus, this would have no effect except confusing the reader and satisfying an incomplete linter check.

Furthermore, the path should be a "safe" one as it points to the YAML to be parsed. There is no reference location to be compared against. Thus, I would say ignore this linter warning as it makes no sense in this context.